### PR TITLE
[IA-2792] [QA-1487] Runtime creation failures after change to GPU/COS images

### PR DESCRIPTION
**Pull request by rtitle**
Problem statement: as part of the COS PR, [jupyter-docker-compose.yaml](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml#L45) references external services like `crypto-detector`, `proxy`, and `welder` in order to set up networking. This is problematic because we don't deploy `crypto-detector` on AoU runtimes, so the `jupyter-docker-compose` crashes. (Note we don't install `crypto-detector` on AoU because AoU requires Dockerhub images, and cryptomining isn't really a problem anyway on AoU.)

I think it would be better if all the docker-compose files were independent of each other, so we could then apply them conditionally in `gce-init.sh` (this is how it was before the COS PR). So in this PR I tried to make each docker-compose file independent. This means we need GCE and Dataproc versions for each of them (because GCE uses a network and Dataproc doesn't); but the changes aren't too bad.

I tested this in a fiab, using Terra UI to create GCE and Dataproc runtimes, and verified all the images launch as expected. I also verified that launching the AoU Dockerhub image works (via Leo swagger), and the cryptomining container does NOT get launched as expected.

Also running automation tests, but hopefully they will pass.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [inline http/src/main/resources/init-resources/gce-init.sh:324]: This was added in a prior PR, but not really needed

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [inline http/src/main/resources/init-resources/gce-init.sh]: I noticed some directories owned by root; not exactly sure why, but figured this is a good idea regardless

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [inline http/src/main/resources/init-resources/gce-init.sh]: I'm not sure why this sleep is needed, and it makes me nervous. Without it, the `run-jupyter.sh` script was not actually taking effect, making me think there's some race condition. Will keep investigating this, but wanted to get a PR out

File: http/src/main/resources/reference.conf
Comment (rtitle) [inline http/src/main/resources/reference.conf:452]: `network-gce.yaml` does not exist in Leo and is not referenced elsewhere, so removed it

**Review by kyuksel — APPROVED** [review]
LGTM  👍🏼   pending integration tests passing and fixing Trivy vulnerabilities.
It may also be good for @Qi77Qi to take a look when she's back.

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [reply http/src/main/resources/init-resources/gce-init.sh]: Think I figured this out, removed the sleep

**Comment by rtitle** [comment]
Thanks @kyuksel! Yeah I feel pretty good based on manual testing -- will confirm automation tests and `multi-test` as well. (I also was able to remove the `sleep` 🎉 )

Yeah weird trivy is failing, the Leo Dockerfile hasn't changed. Will look more..

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest, sbt breakage

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
Yay trivy resolved

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
I think this PR is good but I'm trying to address some test flakiness before merge. In the most recent run 1/4 tests failed with:
```
org.broadinstitute.dsde.workbench.leonardo.runtimes.RuntimeCreationDiskSpec.create runtime and attach an existing persistent disk
```

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
4/4 passes after my last commit! Doing some final manual testing of the aou image, then I will merge this I think and re-enable Argo syncs in dev.

File: http/src/main/resources/init-resources/gce-init.sh
Comment (Qi77Qi) [inline http/src/main/resources/init-resources/gce-init.sh:472]: hmm...this will not work for new jupyter images since the user won't be the same...what was the reason adding `-u`?

File: http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml
Comment (Qi77Qi) [inline http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml:40]: hmm...I did start with duplicating `welder-docker-compose` and `proxy-docker-compose`, but figured it'd be nicer if we can just have one `-gce` and non gce compose file, and have separate network seting appropriately. Since AOU complicates things a bit, I was thinking we can just have a separate `crypto-detector-docker-compose.yaml` that we attach conditionally

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [reply http/src/main/resources/init-resources/gce-init.sh:472]: Ah I forgot the user is different in the new images. I don't think `-u` is really needed, I was trying to fix test flakiness (thought it might be starting as root somehow)

File: http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml
Comment (rtitle) [reply http/src/main/resources/init-resources/jupyter-docker-compose-gce.yaml:40]: I thought it was better if each `*-docker-compose.yaml` file was independent and didn't reference each other. That way we can apply them conditionally without worrying about dependencies. We already have separate GCE and non-GCE `clusterResources` in `reference.conf` so there isn't much extra templating needed.

File: http/src/main/resources/init-resources/gce-init.sh
Comment (rtitle) [reply http/src/main/resources/init-resources/gce-init.sh:472]: Probably also need to fix the `chown` commands above (again, not sure that is really needed, was trying to fix test flakiness but wasn't really successful).

**Pull request merged**
